drivers/timer/mcux_os_timer.c

```diff
@@ -27,13 +29,26 @@
 static struct k_spinlock lock;
 static uint64_t last_count;
 static OSTIMER_Type *base;
+/* Total cycles of the timer compensated to include the time lost in "sleep/deep sleep" modes.
+ * This maintains the timer count to account for the case if the OS Timer is reset in
+ * certain deep sleep modes and the time elapsed when it is powered off.
+ */
+static uint64_t cyc_sys_compensated;
```

Comment:
A few general questions for my own understanding:
- have we done any long term testing with this change, or testing with an external time reference (IE toggling a pin while watching it with an oscilloscope)?
- when would `cyc_sys_compensated` be reset? Maybe I'm missing something here, but wouldn't we reset this value whenever we announce progress to the kernel?

Reply:
@mmahadevan108 ping here- looking at this again, I think I follow the code better now. Essentially, as I understand it:
- `cyc_sys_compensated` will never be reset. Since the OSTimer is 64 bits, we simply use `cyc_sys_compensated` to track cycles "lost" during the sleep period of the system, and never have to worry about resetting it

I'd still like to know if we've verified a case like the following:
- set a long kernel timer (so the system will enter into idle mode and use the counter)
- toggle a pin when the timer fires
- verify that the pin toggles like we expect using an external reference (IE a signal analyzer or oscilloscope)

Reply:
the kernel/timer/timer_api test does this, it calculates the timer interval, and I set to 100s which works fine.